### PR TITLE
Update VJP for softmaxCrossEntropy to account for a call to mean().

### PR DESCRIPTION
**Pull request by pschuh**

### CODE REVIEW

File: Sources/DeepLearning/Loss.swift [hunk 1]
```diff
@@ -45,7 +45,8 @@ func _vjpSoftmaxCrossEntropy<Scalar: TensorFlowFloatingPoint>(
     logits: Tensor<Scalar>, labels: Tensor<Int32>
 ) -> (Tensor<Scalar>, (Tensor<Scalar>) -> Tensor<Scalar>) {
     let (loss, grad) = Raw.sparseSoftmaxCrossEntropyWithLogits(features: logits, labels: labels)
-    return (loss.mean(), { v in v * grad })
+    let nBatches = Tensor<Scalar>(logits.shapeTensor[Int32(0)])
```
Comment (dan-zheng) [inline Sources/DeepLearning/Loss.swift]:
```suggestion
    let batchSize = Tensor<Scalar>(logits.shapeTensor[Int32(0)])
```

File: Sources/DeepLearning/Loss.swift [hunk 2]
```diff
@@ -45,7 +45,8 @@ func _vjpSoftmaxCrossEntropy<Scalar: TensorFlowFloatingPoint>(
     logits: Tensor<Scalar>, labels: Tensor<Int32>
 ) -> (Tensor<Scalar>, (Tensor<Scalar>) -> Tensor<Scalar>) {
     let (loss, grad) = Raw.sparseSoftmaxCrossEntropyWithLogits(features: logits, labels: labels)
-    return (loss.mean(), { v in v * grad })
+    let batchSize = Tensor<Scalar>(logits.shapeTensor[Int32(0)])
```
Comment (rxwei) [inline Sources/DeepLearning/Loss.swift]:
```suggestion
    let batchSize = Tensor<Scalar>(logits.shapeTensor[0])
```

**Comment by marcrasi** [comment]
I think I see the problem with the tests. They got fixed in `master` but not `tensorflow-0.2`. I'll just merge this.

**Comment by marcrasi** [comment]
CI on https://github.com/tensorflow/swift-apis/pull/43 will confirm whether my hasty merge was actually the correct thing to do.
